django/contrib/admindocs/views.py

```diff
@@ -411,8 +412,9 @@ def simplify_regex(pattern):
     """
     pattern = replace_named_groups(pattern)
     pattern = replace_unnamed_groups(pattern)
-    # clean up any outstanding regex-y characters.
-    pattern = pattern.replace('^', '').replace('$', '').replace('?', '')
+    # Remove any outstanding regex metacharacters.
+    pattern = re.sub(r'[?^$+*]', r'', pattern)
+    pattern = re.sub(r'((?:^|[^\\])(?:\\\\)*)\\[bBAZ]', r'\1', pattern)
```

Comment:
These replacements needed to be split up because `r'^\A(?P<slug>\w+)\Z'` would produce `/^<slug>` instead of `/<slug>` as the `^` would be caught by `[^\\]` and substituted back in.